src/forklift/core.py

```diff
@@ -398,8 +398,14 @@ def _is_naughty_field(field):
     '''
     #: global id's do not export to file geodatabase
     #: removes objectid_ which is created by geoprocessing tasks and wouldn't be in destination source
-    #: TODO: Deal with possibility of OBJECTID_* being the OIDFieldName
-    return 'SHAPE' in field.upper() or field.upper() in ['GLOBAL_ID', 'GLOBALID'] or field.startswith('OBJECTID')
+    skip_fields = ['GLOBAL_ID', 'GLOBALID']
+
+    if describe is not None:
+        for prop in ['shapeFieldName', 'lengthFieldName', 'OIDFieldName']:
+            if prop in describe:
+                skip_fields.append(describe[prop].upper())
+    field = field.upper()
+    return 'SHAPE' in field or field in skip_fields or field.startswith('OBJECTID')
```

Comment:
**thought (non-blocking)**: do we need the objectid thing still if we use the `OIDFieldName`? Is this here for tables with multiple versions of an OID?

Reply:
Yes. There are some tables that have junk `OBJECTID_1` fields that are not the official oid field.